pkg/oci/layout/write.go

```diff
@@ -27,41 +25,29 @@ import (
 
 // WriteSignedImage writes the image and all related signatures, attestations and attachments
 func WriteSignedImage(path string, si oci.SignedImage) error {
-	// First, write the image
-	if err := write(path, imagePath, si); err != nil {
-		return errors.Wrap(err, "writing image")
+	// First, write an empty index
+	layoutPath, err := layout.Write(path, empty.Index)
+	if err != nil {
+		return err
+	}
+	// write the image
+	if err := appendImage(layoutPath, si, imageAnnotation); err != nil {
+		return errors.Wrap(err, "appending signed image")
 	}
+	// write the signatures
 	sigs, err := si.Signatures()
 	if err != nil {
 		return errors.Wrap(err, "getting signatures")
 	}
-	if err := write(path, signaturesPath, sigs); err != nil {
-		return errors.Wrap(err, "writing signatures")
+	if err := appendImage(layoutPath, sigs, sigsAnnotation); err != nil {
```

Comment:
If the resulting index were pushed, how would it run?

cc @jonjohnsonjr

Reply:
Generally clients take the first entry that matches their criteria (platform).

This might be a reasonable knob to expose when pushing to a registry? By default, just push all the entries in `index.json`, but optionally include that `index.json` as a top-level index for everything related to the artifact. I don't know how this is intended to be used.

Reply:
for `cosign load` I was planning on pushing everything in `index.json` for now and not including the `index.json` itself so this effectively is the same as running `cosign copy` -- i think it would be cool to optionally include pushing `index.json` as a feature for later (although running into some errors with it atm that would need to be addressed)